lrlex/src/lib/ctbuilder.rs

```diff
@@ -495,10 +497,49 @@ where
                                 .join("\n")
                         })?;
                 let lex_flags = lexerdef.lex_flags().cloned();
-                (Box::new(lexerdef), lex_flags.unwrap())
+                (lexerdef, lex_flags.unwrap())
             }
         };
 
+        if let Some(ref lrcfg) = self.lrpar_config {
+            let mut lexerdef = lexerdef.clone();
+            let mut ctp = CTParserBuilder::<LexerTypesT>::new().inspect_rt(Box::new(
+                move |yacc_header, rtpb, rule_ids_map, grm_path| {
+                    let owned_map = rule_ids_map
+                        .iter()
+                        .map(|(x, y)| (&**x, *y))
+                        .collect::<HashMap<_, _>>();
+                    lexerdef.set_rule_ids(&owned_map);
+                    yacc_header.mark_used(&"test_files".to_string());
+                    let test_glob = yacc_header.get("test_files");
+                    match test_glob {
+                        Some(HeaderValue(_, Value::Setting(Setting::String(test_files, _)))) => {
+                            let path_joined = grm_path.parent().unwrap().join(test_files);
+                            for path in
+                                glob(&path_joined.to_string_lossy()).map_err(|e| e.to_string())?
+                            {
+                                let path = path?;
+                                let mut file = File::open(&path)?;
+                                let mut input = String::new();
```

Comment:
Can these lines become something like `let input = fs::read_to_string(&path)`?

Reply:
done in [c7a4983](https://github.com/softdevteam/grmtools/pull/559/commits/c7a49836b8a2aac847927367c28eb135d235f571)